ras_party/controllers/respondent_controller.py

```diff
@@ -70,35 +72,37 @@ def get_respondent_by_id(respondent_id, session):
 
 
 @with_db_session
-def delete_respondent_by_id(party_uuid, session):
+def delete_respondent_by_email(email, session):
     """
-    Delete a Respondent by its Party ID
+    Delete a Respondent by its email
     On success it returns None, on failure will raise one of many different exceptions
     :param party_uuid: Id of Respondent to delete
     :type party_uuid: str
     """
-    logger.info("Starting to delete respondent", party_uuid=party_uuid)
-    try:
-        uuid.UUID(party_uuid)
-    except ValueError:
-        logger.info("party_uuid value is not a valid UUID", party_uuid=party_uuid)
-        raise BadRequest(f"'{party_uuid}' is not a valid UUID format for property 'party_uuid'")
+    logger.info("Starting to delete respondent", email=obfuscate_email(email))
 
     # We need to get the respondent to make sure they exist, but also because the id (not the party_uuid...for
     # some reason) of the respondent is needed for the later deletion steps.
-    respondent = query_respondent_by_party_uuid(party_uuid, session)
-    if not respondent:
-        logger.info("Respondent with party_uuid does not exist", party_uuid=party_uuid)
-        raise NotFound("Respondent with party_uuid does not exist")
+    try:
+        respondent = query_single_respondent_by_email(email, session)
+    except NoResultFound:
+        logger.error("Respondent with email does not exist", email=obfuscate_email(email))
+        raise NotFound("Respondent with email does not exist")
+    except MultipleResultsFound:
+        logger.error("Multiple respondents found for email", email=obfuscate_email(email))
+        raise UnprocessableEntity("Multiple users found, unable to proceed")
 
-    logger.info("Found respondent", party_uuid=str(respondent.party_uuid), id=respondent.id)
+    logger.info("Found respondent", email=respondent.email_address, party_uuid=respondent.party_uuid, id=respondent.id)
```

Comment:
This log line either cannot log the email or the email needs to be obfuscated

Reply:
Good catch!  Thought I'd got them all